src/xbot/common/command/scripted/CommandScriptInterface.java

```diff
@@ -0,0 +1,85 @@
+package xbot.common.command.scripted;
+
+import org.apache.log4j.Logger;
+
+import edu.wpi.first.wpilibj.command.Command;
+import xbot.common.command.BaseCommand;
+
+/**
+ * An object which can be exposed to a running script to wrap a
+ * command on the scheduler.
+ *
+ */
+public class CommandScriptInterface {
+    static Logger log = Logger.getLogger(CommandScriptInterface.class);
+    
+    private Command wrappedCommand;
+    private ScriptedCommand parentCommand;
+
+    public CommandScriptInterface(Command wrappedCommand, ScriptedCommand parentCommand) {
+        this.wrappedCommand = wrappedCommand;
+        this.parentCommand = parentCommand;
+    }
+    
+    public boolean canWaitForCompletion() {
+        return isBaseCommand();
+    }
+    
+    public boolean canCheckIsFinished() {
+        return isBaseCommand();
+    }
+    
+    public boolean isBaseCommand() {
+        return this.wrappedCommand instanceof BaseCommand;
+    }
+    
+    public boolean isRunning() {
+        return wrappedCommand.isRunning();
+    }
+
+    public boolean isFinished() {
+        if(!(this.wrappedCommand instanceof BaseCommand)) {
+            log.error("isFinished called on command which doesn't inherit from BaseCommand!"
+                    + " Unable to check if command has finished.");
+            
+            return false;
+        }
+        
+        return ((BaseCommand)wrappedCommand).isFinished();
+    }
+    
+    public void waitForCompletion(long timeoutMillis) {
+        if(!(this.wrappedCommand instanceof BaseCommand)) {
+            log.error("waitForCompletion called on command which doesn't inherit from BaseCommand!"
+                    + " Unable to wait for command to complete.");
+            
+            return;
+        }
+        
+        ScriptedCommandCompletionLock lock = new ScriptedCommandCompletionLock((BaseCommand)wrappedCommand);
+        parentCommand.addCompletionLock(lock);
+
+        try {
+            long startTime = System.currentTimeMillis();
+            synchronized (lock) {
+                if(timeoutMillis <= 0) {
+                    log.debug("Waiting indefinitely for lock notification");
+                    lock.wait();
+                }
+                else {
+                    log.debug("Waiting with a timeout of " + timeoutMillis + "ms");
+                    lock.wait(timeoutMillis);
+                }
+            }
+            
+            log.debug("Wait completed (waited " + (System.currentTimeMillis() - startTime) + "millis)");
```

Comment:
There isn't really a way to do that (as far as I can tell); the `lock.wait` method doesn't return anything indicating what happened. I could compare my timer value to the timeout value that I passed to `lock.wait`, but that is prone to errors and isn't an option for reliable results.